### PR TITLE
Give random NPCs a new story type to offer breadcrumbs to finding other NPCs

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
Content "Adds new stories to random NPCs, making them more able to help you find other NPCs."

#### Purpose of change
Mapgen can make it very difficult to find faction camps and static NPCs. Giving random spawning NPCs some knowledge of the world would help make things feel more connected, and improve player ability to find story quests and more.

#### Describe the solution
This adds a new random background story category for NPCs, "other survivors stories" or BOSS. These stories are randomly assigned on NPC creation just like BGSS stories are. Many of them will have a chance to include a mission leading to various interesting static NPCs and locations, or to their reanimated remains.

In the process due to clutter I moved the existing background stories to a new subfolder.

#### Describe alternatives you've considered
We could integrate these into existing stories and things, but I would say that's another thing we can do rather than something to do instead.

#### Testing
wip

#### Additional context
At present, only your followers can give you this advice. We should look into making this something cautious non-follower NPCs would talk to you about as well.

@Maleclypse started this and I'm helping him out.

### CODE REVIEW

File: data/json/npcs/Backgrounds/OtherSurvivorStories/Lapin_01.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  {
+    "id": "BOSS_Lapin_01_Story1",
+    "type": "talk_topic",
+    "dynamic_line": "Actually, yeah.  It's a long shot but we could follow it if you like.  I ran into another survivor once - don't think she made it, she was heading to a nearby mall, and that place was overrun.  She mentioned seeing a guy taking care of his cabin in the woods though.  Said she thought about checking on him, but that was when things were all going nuts, and she decided against it.  I got some directions and never followed through on it.",
+    "responses": [
+      {
+        "text": "Let's see how to get there and go check on the guy.",
+        "topic": "BOSS_Lapin_01_Story2",
+        "condition": { "npc_has_trait": "BOSS_No_Others_02" },
+      {
```
Comment (Xaleth) [inline data/json/npcs/Backgrounds/OtherSurvivorStories/Lapin_01.json:11]: I think this brace should be flipped

Comment (Xaleth) [inline data/json/npcs/Backgrounds/OtherSurvivorStories/Lapin_01.json:11]:
```suggestion
      },
```

File: data/json/npcs/Backgrounds/OtherSurvivorStories/Lapin_01.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        "topic": "BOSS_Lapin_01_Story2",
+        "condition": { "npc_has_trait": "BOSS_No_Others_02" },
+      },
+        "text": "<done_conversation>",
+        "topic": "TALK_FRIEND"
+      },
+      { "text": "<end_talking>", "topic": "TALK_DONE" }
+    ]
+  }
+]
```
Comment (LyleSY) [inline data/json/npcs/Backgrounds/OtherSurvivorStories/Lapin_01.json:18]: `[
  {
    "id": "BOSS_Lapin_01_Story1",
    "type": "talk_topic",
    "dynamic_line": "Actually, yeah.  It's a long shot but we could follow it if you like.  I ran into another survivor once - don't think she made it, she was heading to a nearby mall, and that place was overrun.  She mentioned seeing a guy taking care of his cabin in the woods though.  Said she thought about checking on him, but that was when things were all going nuts, and she decided against it.  I got some directions and never followed through on it.",
    "responses": [
      {
        "text": "Let's see how to get there and go check on the guy.",
        "topic": "BOSS_Lapin_01_Story2",
        "condition": { "npc_has_trait": "BOSS_No_Others_02" }
      },
      { "text": "<done_conversation>", "topic": "TALK_FRIEND" },
      { "text": "<end_talking>", "topic": "TALK_DONE" }
    ]
  }
]
`

Comment (LyleSY) [inline data/json/npcs/Backgrounds/OtherSurvivorStories/Lapin_01.json:18]:
```suggestion
[
  {
    "id": "BOSS_Lapin_01_Story1",
    "type": "talk_topic",
    "dynamic_line": "Actually, yeah.  It's a long shot but we could follow it if you like.  I ran into another survivor once - don't think she made it, she was heading to a nearby mall, and that place was overrun.  She mentioned seeing a guy taking care of his cabin in the woods though.  Said she thought about checking on him, but that was when things were all going nuts, and she decided against it.  I got some directions and never followed through on it.",
    "responses": [
      {
        "text": "Let's see how to get there and go check on the guy.",
        "topic": "BOSS_Lapin_01_Story2",
        "condition": { "npc_has_trait": "BOSS_No_Others_02" }
      },
      { "text": "<done_conversation>", "topic": "TALK_FRIEND" },
      { "text": "<end_talking>", "topic": "TALK_DONE" }
    ]
  }
]
```

**Comment by I-am-Erk** [comment]
Closing this as i have enough on my plate right now, I will try to get back to it as the infrastructure to finish it is now in place.

**Pull request closed without merging**